local-volume/onetimecontroller/controller.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/common"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func updatePVAlphaAnn(client *kubernetes.Clientset, pv *v1.PersistentVolume) error {
+	if len(pv.Annotations) > 0 && pv.Annotations[v1.AlphaStorageNodeAffinityAnnotation] != "" {
+		var affinity v1.NodeAffinity
+		err := json.Unmarshal([]byte(pv.Annotations[v1.AlphaStorageNodeAffinityAnnotation]), &affinity)
+		if err != nil {
+			return err
+		}
+		pvClone := pv.DeepCopy()
+		delete(pvClone.Annotations, v1.AlphaStorageNodeAffinityAnnotation)
+		pvClone.Spec.NodeAffinity = &v1.VolumeNodeAffinity{
+			Required: affinity.RequiredDuringSchedulingIgnoredDuringExecution,
+		}
+
+		_, err = client.CoreV1().PersistentVolumes().Update(pvClone)
+		if err != nil {
+			return err
```

Comment:
Similarly for errors.IsNotFound() you could log a message saying you are skipping the PV as it seems to have been deleted. In general this would reassure the user that such errors are not really errors, but anticipated changes in PV state.

Reply:
will do